core/shared/src/main/scala/com/ccadllc/cedi/dtrace/TraceSystem.scala

```diff
@@ -62,6 +66,29 @@ object TraceSystem {
     def description: String
   }
 
+  final case class Data(
+    identity: Map[Data.Key, Data.Value],
+    meta: Map[Data.Key, Data.Value]) {
+    val combined: Map[Data.Key, Data.Value] = identity ++ meta
+    val identityStrMap: Map[String, String] = stringified(identity)
+    val metaStrMap: Map[String, String] = stringified(meta)
+    val combinedStrMap: Map[String, String] = stringified(combined)
+    val description: String = s"identity: $identityStrMap, meta: $metaStrMap"
+
+    private def stringified(data: Map[Data.Key, Data.Value]): Map[String, String] =
+      data.map { case (k, v) => k.value -> k.value }
+  }
+  object Data {
+    final class Key(val value: String) extends AnyVal
```

Comment:
Is the intent to make the default new constructor private?

Reply:
No, just a convenience - no need to make it private since there’s no validation (they’re just new types). Then again making them value types versus case classes isn’t worth it in this case (in most cases really)since it doesn’t save on boxing as they’re only used in context of the collection.And even if they did, it’s a small collection created once. I’ll update to make case classes.